lua/copilot/health.lua

```diff
@@ -0,0 +1,93 @@
+local M = {}
+
+local api = require("copilot.api")
+local auth = require("copilot.auth")
+local c = require("copilot.client")
+local config = require("copilot.config")
+
+local start = vim.health.start or vim.health.report_start
+local ok = vim.health.ok or vim.health.report_ok
+local warn = vim.health.warn or vim.health.report_warn
+local error = vim.health.error or vim.health.report_error
+local info = vim.health.info or vim.health.report_info
+
+function M.check()
```

Comment:
Here's a screenshot of what this current healthcheck setup looks like:
<img width="554" height="461" alt="Screenshot 2025-07-25 at 12 19 47 PM" src="https://github.com/user-attachments/assets/69e71737-89f2-41ee-adf4-39e301d50c4a" />